File: app/src/main/java/com/example/habitproproject/DiasAdapter.java
package com.example.habitproproject;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class DiasAdapter extends RecyclerView.Adapter<DiasAdapter.DiaViewHolder> {

    private List<Dia> listaDias;

    public DiasAdapter(List<Dia> listaDias) {
        this.listaDias = listaDias;
    }

    @NonNull
    @Override
    public DiaViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_dia, parent, false);
        return new DiaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DiaViewHolder holder, int position) {
        Dia dia = listaDias.get(position);
        holder.tvNombreAbreviado.setText(dia.getNombreAbreviado());
        holder.tvNumeroDia.setText(String.valueOf(dia.getNumeroDia()));
    }

    @Override
    public int getItemCount() {
        return listaDias.size();
    }

    static class DiaViewHolder extends RecyclerView.ViewHolder {
        TextView tvNombreAbreviado, tvNumeroDia;

        public DiaViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNombreAbreviado = itemView.findViewById(R.id.tvNombreAbreviado);
            tvNumeroDia = itemView.findViewById(R.id.tvNumeroDia);
        }
    }
}

File: app/src/main/java/com/example/habitproproject/DiasAdapter.kt
package com.example.habitproproject

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class DiasAdapter(private val listaDias: List<Dia>) : RecyclerView.Adapter<DiasAdapter.DiaViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DiaViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_dia, parent, false)
        return DiaViewHolder(view)
    }

    override fun onBindViewHolder(holder: DiaViewHolder, position: Int) {
        val dia = listaDias[position]
        holder.tvNombreAbreviado.text = dia.nombreAbreviado
        holder.tvNumeroDia.text = dia.numeroDia.toString()
    }

    override fun getItemCount(): Int {
        return listaDias.size
    }

    class DiaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvNombreAbreviado: TextView = itemView.findViewById(R.id.tvNombreAbreviado)
        val tvNumeroDia: TextView = itemView.findViewById(R.id.tvNumeroDia)
    }
}